functions/index.js

```diff
@@ -0,0 +1,65 @@
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+
+admin.initializeApp(functions.config().firebase);
+const db = admin.firestore();
+
+const REQUEST_INVITE = 'REQUEST_INVITE';
+
+function createMemberFromOperation(operation) {
+    const {
+        applied,
+        block_at,
+        block_seq,
+        created_at,
+        creator_uid,
+        creator_mid,
+        data,
+        op_code,
+        op_seq
+    } = operation;
+
+    if (applied || op_code !== REQUEST_INVITE) {
```

Comment:
Need more data validation as at this point our `firestore.rules` are quite simple. We should check every field coming from a (possibly malicious) client. Eg that block_at block_seq op_seq are all null, that created_at really is a date, that full_name and creator_mid are strings and they match, etc

Reply:
After discussion: needs more thought, see issue #61.